### ISSUE TITLE
Axion-scan commands not working

### ISSUE THREAD

**Issue by daai1** [pry0cc/axiom#548]
@pry0cc @0xtavian Please help i want to run this single template and i'm not able to run.

Also i have my custom templates in this path /root/custom-templates how can i run my custom templates

axiom-scan ~/target/subs_httpx -m nuclei --nuclei-templates /root/nuclei-templates/cves/2020/CVE-2020-0XXX.yaml -o nuclei.txt

axiom-scan ~/target/subs_httpx -m nuclei -wL /root/custom-templates/CVE-2020-0XXX.yaml -o nuclei.txt

**Comment by 0xtavian** [comment]
To upload a folder of nuclei templates use `—nuclei-templates`. To upload a single template use `-wL`. If you are still having issue post the command and terminal output.

**Comment by daai1** [comment]
~ axiom-scan ~/target/target_subs_httpx -m nuclei --nuclei-templates /root/nt -o nuclei_nt       
              _
  ____ __  __(_)___  ____ ___        ______________ _____
 / __ `/ |/_/ / __ \/ __ `__ \______/ ___/ ___/ __ `/ __ \
/ /_/ />  </ / /_/ / / / / / /_____(__  ) /__/ /_/ / / / /
\__,_/_/|_/_/\____/_/ /_/ /_/     /____/\___/\__,_/_/ /_/

                                    @pry0cc
                                 & @0xtavian

creating scan working directory at : /home/op/scan/nuclei+1651301765/
uploading local folder ( nuclei-templates ) : /root/nt to /home/op/scan/nuclei+1651301765/nt...
custom folder uploaded successfully!
module: [ nuclei ] | module args: [  ] | input: [ 10 lines ] |
instances:  5  [ bounty01 bounty02 bounty03 bounty04 bounty05 ] |
command: [ /home/op/go/bin/nuclei -silent -update ; cat input | /home/op/go/bin/nuclei -t /home/op/scan/nuclei+1651301765/nt  -o output ] | ext: [ txt ] | threads: [ null ]
spliting and distributing input file...
[ OK ]

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   2.6.9

                projectdiscovery.io

[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[INF] Your current nuclei-templates v8.9.7 are outdated. Latest is v8.9.8
[INF] Downloading latest release...

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   2.6.9

                projectdiscovery.io

[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[INF] Your current nuclei-templates v8.9.7 are outdated. Latest is v8.9.8
[INF] Downloading latest release...

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   2.6.9

                projectdiscovery.io

[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[INF] Your current nuclei-templates v8.9.7 are outdated. Latest is v8.9.8
[INF] Downloading latest release...

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   2.6.9

                projectdiscovery.io

[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[INF] Your current nuclei-templates v8.9.7 are outdated. Latest is v8.9.8
[INF] Downloading latest release...
                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   2.6.9

                projectdiscovery.io

[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.

[INF] Your current nuclei-templates v8.9.7 are outdated. Latest is v8.9.8
[INF] Downloading latest release...
[INF] Successfully updated nuclei-templates (v8.9.8) to /home/op/nuclei-templates. GoodLuck!
[INF] Successfully updated nuclei-templates (v8.9.8) to /home/op/nuclei-templates. GoodLuck!
[INF] Using Nuclei Engine 2.6.9 (latest)
[INF] Using Nuclei Templates 8.9.8 (latest)
[FTL] Could not run nuclei: no valid templates were found
[INF] Using Nuclei Engine 2.6.9 (latest)
[INF] Using Nuclei Templates 8.9.8 (latest)
[FTL] Could not run nuclei: no valid templates were found
[INF] Successfully updated nuclei-templates (v8.9.8) to /home/op/nuclei-templates. GoodLuck!
[INF] Using Nuclei Engine 2.6.9 (latest)
[INF] Using Nuclei Templates 8.9.8 (latest)
[FTL] Could not run nuclei: no valid templates were found
[INF] Successfully updated nuclei-templates (v8.9.8) to /home/op/nuclei-templates. GoodLuck!
[INF] Using Nuclei Engine 2.6.9 (latest)
[INF] Using Nuclei Templates 8.9.8 (latest)
[FTL] Could not run nuclei: no valid templates were found
[INF] Successfully updated nuclei-templates (v8.9.8) to /home/op/nuclei-templates. GoodLuck!

@pry0cc @0xtavian I'm getting error like [FTL] Could not run nuclei: no valid templates were found.

**Comment by 0xtavian** [comment]
You can ssh into an instance, cd into the scan working directory, check if the templates were uploaded and try running the command manually?

**Comment by absane** [comment]
Wanted to jump in to say that I am also having issues with axiom-scan. Basically, I can't get it to work for nmap. I've installed Axiom twice now, one within a VM and another with Docker. IN both installs, I had issues with Interlace not working with Python 3.10, which I think I fixed. When I run this command, it seems that everything worked:

`axiom-scan test.txt -m nmap -sT -p- -sV -v --open -oA nmap -sC`

The output claims that Axiom is doing something, but after it exists I get no scan results. When I SSH to the Axiom server and check the scan folder, all; the files exist but are empty, including the "command" file that I would expect to have the commands I sent.

When I try the nmapx module, I have better luck actually seeing the command get executed on the server. However, after scanning a few ports the nmap process just hangs, using no CPU cycles.

I've got no idea where to go from here.

**Comment by 0xtavian** [comment]
@absane can you show me the nmap module?

**Comment by 0xtavian** [comment]
Are you seeing any errors running `axiom-exec id`? If so you might need to follow this temp workaround if using axiom docker (https://github.com/pry0cc/axiom/issues/555). If you are running Ubuntu 20.04 for your axiom controller, you dont need the temp workaround. What is your base OS?

**Comment by absane** [comment]
Running commands directly with `axiom-exec id` work just find. So, I can manually kick off nmap that way.

Base OS for the VM I tried was Kali Linux. For the Docker, I am running that in Windows. Both give the same exact problems. Both are also using the Axiom as I did a fresh install a few hours ago.

I found the workaround to the Interlace issue I was initially having at this [issue comment](https://github.com/pry0cc/axiom/issues/555#issuecomment-1133745166).

I just now edited `.axiom/modules/nmap.json` and I think that actually solved my problem, in combination with the Interlace work around.  I am trying different scan options and they all seem to work now. When I did the --debug option, I did see an error at the bottom stating that the nmap files could not be created/accessed, which is what broke the whole thing. It's clear as to why now.

I don't know why I didn't think of it before, but also I think maybe in the past the `-oA` options was supported, indirectly or my memory is misleading me. Either way, it seems to work now.

Much appreciated!
